### DISCUSSION TITLE
polar decomposition

### DISCUSSION THREAD

**Discussion by jgenee** [thelfer/tfel#188] [Q&A] [answered]
Hi everyone,

I was wondering if polar_decomposition function is still implemented within TFEL ? I found some documentation about it in mfront website (cf. http://tfel.sourceforge.net/tensors.html), though I cannot find the directory where the hxx file is stored.. (in order to include it into my mfront file).

Thanks in advance.

**Comment by jgenee** [comment, marked as answer]
Asking help too fast.. I just find the solution using :

`tensor<N, real> R;
  stensor<N, strain> U;
  polar_decomposition(R, U, F);`

I was first delcaring R and U as auxiliary state variables, but it did not seem to work then when calling them in polar decomposition function.

**Reply by thelfer** [reply]
@jgenee, I am quite surprised by your remark. This compiles:

```cxx
@DSL DefaultFiniteStrainDSL;
@Behaviour Test;
@Author Thomas Helfer;
@Date 11 / 05 / 2022;

@AuxiliaryStateVariable Tensor R;
@AuxiliaryStateVariable StrainStensor U;

@Integrator {
  polar_decomposition(R, U, F1);
}
```
